classes/actions.php

```
@@ -262,8 +270,15 @@ public static function duplicate_to_course(array $modules, int $targetcourseid,
         // Let order of mods in a section be mod1, mod2, mod3, mod4, mod5. If we duplicate mod2, mod4, the order afterwards will be
         // mod1, mod2, mod3, mod4, mod5, mod2(dup), mod4(dup).
         $duplicatedmods = [];
+        $sourceformat = course_get_format($sourcecourseid);
+        $sourcesectionsrestricted = massactionutils::get_restricted_sections($sourcecourseid, $sourceformat->get_format());
         foreach ($idsincourseorder as $cmid) {
-            $duplicatedmod = massactionutils::duplicate_cm_to_course($targetmodinfo->get_course(), $sourcemodinfo->get_cm($cmid));
+            $sourcecm = $sourcemodinfo->get_cm($cmid);
+            // Not duplicated if the section is restricted.
+            if (in_array($sourcecm->sectionnum, $sourcesectionsrestricted)) {
+                continue;
```

Should we throw an exception of sectionrestricted here?